File: 03_Linear_equations/Scripts/forwardSub.m
function [x] = forwardSub(A,b)

% Check the matrix - should be square
[N,M] = size(A);
if (N ~= M)
    error('Matrix A in forwardSub should be square');
end

% Assign b to x
x=b;

% Perform forward substitution
for row=1:N
    for j =1:row-1
        x(row)=x(row)-A(row,j)*x(j);
    end
    x(row)=x(row)/A(row,row);
end

File: scripts/used_by_students/Scripts_linear_systems_1/forwardSub.py
import numpy as np

def forwardSub(A,b):
    
    # Check the matrix - should be square
    N,M = A.shape
    if(N!=M):
        print('Matrix A in forwardSub should be square')
        return
    
    # Assign b to x
    x = b
    
    # Perform forward substitution
    for row in range(N):
        for j in range(row):
            x[row] = x[row] - A[row,j] * x[j]
        x[row] = x[row] / A[row,row]
        
    return x